virtcontainers/sandbox.go

```diff
@@ -1357,17 +1478,49 @@ func (s *Sandbox) HotplugAddDevice(device api.Device, devType config.DeviceType)
 func (s *Sandbox) HotplugRemoveDevice(device api.Device, devType config.DeviceType) error {
 	switch devType {
 	case config.DeviceVFIO:
-		vfioDevice, ok := device.(*drivers.VFIODevice)
+		vfioDevices, ok := device.GetDeviceDrive().([]*config.VFIODrive)
 		if !ok {
 			return fmt.Errorf("device type mismatch, expect device type to be %s", devType)
 		}
-		return s.hypervisor.hotplugRemoveDevice(*vfioDevice, vfioDev)
+		removedDev := []*config.VFIODrive{}
+		var err error
+		defer func() {
+			// if err happens,roll back and remove added device!
+			if err != nil {
+				for _, dev := range removedDev {
+					if rollbackErr := s.hypervisor.hotplugAddDevice(dev, vfioDev); rollbackErr != nil {
```

Comment:
comment says `// if err happens,roll back and remove added device!`, but here you are adding devices ?

Reply:
The comment is wrong, thanks! It's a typical copy-and-paste error :wink: